test/battle/move_effect/protect.c

```diff
@@ -0,0 +1,369 @@
+        if (hp == 1) {
+            TURN { MOVE(opponent, MOVE_SPIKY_SHIELD); MOVE(player, usedMove); SEND_OUT(player, 1); }
+        } else {
+            TURN { MOVE(opponent, MOVE_SPIKY_SHIELD); MOVE(player, usedMove); }
+        }
+        TURN {}
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_SPIKY_SHIELD, opponent);
+        MESSAGE("Foe Wobbuffet protected itself!");
+        NOT ANIMATION(ANIM_TYPE_MOVE, usedMove, player);
+        MESSAGE("Foe Wobbuffet protected itself!");
+        NOT HP_BAR(opponent);
+        if (usedMove == MOVE_TACKLE) {
+            HP_BAR(player, maxHp / 8);
+            if (hp == 1) {
+                MESSAGE("Wobbuffet fainted!");
+                MESSAGE("Go! Wobbuffet!");
+            }
+        }
+    }
+}
+
+SINGLE_BATTLE_TEST("Baneful Bunker poisons pokemon for moves making contact")
+{
+    u16 usedMove = MOVE_NONE;
+
+    PARAMETRIZE {usedMove = MOVE_TACKLE; }
+    PARAMETRIZE {usedMove = MOVE_LEER; }
+    PARAMETRIZE {usedMove = MOVE_WATER_GUN; }
+
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_BANEFUL_BUNKER); MOVE(player, usedMove); }
+        TURN {}
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_BANEFUL_BUNKER, opponent);
+        MESSAGE("Foe Wobbuffet protected itself!");
+        NOT ANIMATION(ANIM_TYPE_MOVE, usedMove, player);
+        MESSAGE("Foe Wobbuffet protected itself!");
+        NOT HP_BAR(opponent);
+        if (usedMove == MOVE_TACKLE) {
+            STATUS_ICON(player, STATUS1_POISON);
+        } else {
+            NOT STATUS_ICON(player, STATUS1_POISON);
+        }
+    }
+}
+
+SINGLE_BATTLE_TEST("Recoil is not applied if battler was protected") // #3319
+{
+    u16 move = MOVE_NONE;
+
+    PARAMETRIZE {move = MOVE_PROTECT; }
+    PARAMETRIZE {move = MOVE_DETECT; }
+    PARAMETRIZE {move = MOVE_KINGS_SHIELD; }
+    PARAMETRIZE {move = MOVE_BANEFUL_BUNKER; }
+    PARAMETRIZE {move = MOVE_SILK_TRAP; }
+    PARAMETRIZE {move = MOVE_OBSTRUCT; }
+    PARAMETRIZE {move = MOVE_SPIKY_SHIELD; }
+
+    GIVEN {
+        ASSUME(gBattleMoves[MOVE_DOUBLE_EDGE].effect == EFFECT_RECOIL_33);
+        PLAYER(SPECIES_RAPIDASH);
+        OPPONENT(SPECIES_BEAUTIFLY);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_TACKLE); MOVE(player, MOVE_TACKLE); }
+        TURN { MOVE(opponent, move); MOVE(player, MOVE_DOUBLE_EDGE); }
+        TURN {}
+    } SCENE {
+        // 1st turn
+        MESSAGE("Foe Beautifly used Tackle!");
+        MESSAGE("Rapidash used Tackle!");
+        // 2nd turn
+        ANIMATION(ANIM_TYPE_MOVE, move, opponent);
+        MESSAGE("Foe Beautifly protected itself!");
+        MESSAGE("Rapidash used Double-Edge!");
+        NOT ANIMATION(ANIM_TYPE_MOVE, MOVE_DOUBLE_EDGE, player);
+        NOT MESSAGE("Rapidash is hit with recoil!");
+    }
+}
+
+SINGLE_BATTLE_TEST("Multi-hit moves don't hit a protected target and fail only once") // #3312
```

Comment:
```suggestion
SINGLE_BATTLE_TEST("Multi-hit moves don't hit a protected target and fail only once")
```

Reply:
Why removing the comment? I added it intentionally that this test is written because of a submitted issue